### PR TITLE
feat: remove commented code

### PR DESCRIPTION
**Pull request by ynsehoornenborg**
This new analyzer check rule [4@111](https://csviewer.tiobe.com/#/ruleset/rule?setid=4T_Jr6-VSX6fp6egDIhGow&status=CHECKED,UNCHECKED&tagid=6u27MkXORKaev0VNuWR_SA&ruleid=oGFi5e5WTrmunRVL60QkYg) of the Philips C# Coding Standard

### CODE REVIEW

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+		private static readonly DiagnosticDescriptor Rule = new(Helper.ToDiagnosticId(DiagnosticIds.RemoveCommentedCode), Title, MessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterSyntaxNodeAction(Analyze, SyntaxKind.CompilationUnit);
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs:30]: Shoudn't we just register on SingleLineCommentTrivia?

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+			var comments = node.DescendantTrivia().Where(trivia => trivia.IsKind(SyntaxKind.SingleLineCommentTrivia));
+			if (!comments.Any())
+			{
+				return;
+			}
+
+			foreach (var location in comments.Where(comment => comment.ToString().EndsWith(";")).Select(node => node.GetLocation()))
+			{
+				var lineNumber = location.GetLineSpan().StartLinePosition.Line + 1;
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs:51]: If there are 20 straight lines of commented out code, we probably shouldn't have 20 findings. Maybe see if the previous line is also a SingleLineTriviaComment and if so exclude? That way we just have a single finding for the chunk.

**Review by bcollamore — APPROVED** [review]
CodeFixer!  (That way the client can bulk clean it up.)

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs (hunk 1 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs:30]: Trivia are not SyntaxNodes, so you can't filter on them. Copyright Analyzer does the same.

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs (hunk 2 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs:51]: Makes sense

Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs:52]: Implemented and tested

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+				return;
+			}
+
+			var comments = node.DescendantTrivia().Where(trivia => trivia.IsKind(SyntaxKind.SingleLineCommentTrivia));
+			if (!comments.Any())
+			{
+				return;
+			}
+
+			int previousViolationLine = -20;
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs:49]: -20? MagicConstant?

**Review by bcollamore — APPROVED** [review]

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs (hunk 3 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs:49]: Got me !

**Review by bcollamore — APPROVED** [review]

**Pull request merged**